AgContext.yaml

```diff
@@ -0,0 +1,149 @@
+
+agcontext_name:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      Agricultural context name
+      string. Required.
+      A string representing the name of the agricultural context.
+
+crop_type:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      General crop type
+      string. Required.
+      A string describing the general cropping scenario (e.g. "root vegetable").
+
+crop_stage:
+  type:
+    anyOf:
+    - type: string
+      enum:
+      - na
+      - 00
+      - 0
+      - 1
+      - 2
+      - 3
+      - 4
+      - 5
+      - 6
+      - 7
+      - 8
+      - 9
+      - 99
+    description: |-
+      BBCH crop development stage
+      "na", "00", "0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "99". Required.
+      One of eleven possible strings representing the BBCH phenological development stage of the crop. If this AgContext is not in a cropping environment, use value "na".
+
+location_continent:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      Continent where AgContext is located
+      string. Required.
+      A string representing the continent where the AgContext is located.
+
+location_country:
+  type:
+    anyOf:
+    - type: string
+    description: |-
+      Country where AgContext is located
+      string. Required.
+      A string representing the country where the AgContext is located.
+
+location_state:
```

Comment:
Is this a useful granularity? Is there some other way of defining coarse-grained region? Even country + postcode? It's worth asking our horticulturalists what level of granularity is useful...

Or just request a lat-long with jitter of up to defined limits permitted for privacy reasons.

We are able to allow users to filter by region while using a more specific lat-long location in the data input and internally.